src/Compilers/CSharp/Portable/Binder/PatternSwitchBinder2.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    // We use a subclass of SwitchBinder for the pattern-matching switch statement until we have completed
+    // a totally compatible implementation of switch that also accepts pattern-matching constructs.
+    internal partial class PatternSwitchBinder2 : SwitchBinder
```

Comment:
>internal [](start = 4, length = 8)

This class is just a placeholder for now, not exercised. The next PR will fill this out (e.g. computing the DAG and subsumption)

Reply:
> This class is just a placeholder for now, not exercised. The next PR will fill this out (e.g. computing the DAG and subsumption)

Is this accurate? There is a code in SwitchBinder.cs that creates an instance of this type.

---
In reply to: [151486979](https://github.com/dotnet/roslyn/pull/23209#discussion_r151486979) [](ancestors = 151486979)